tests/acl/test_acl_l3v4v6.py

```diff
@@ -0,0 +1,1224 @@
+
+    t2_information = {
+        "upstream_port_ids": upstream_port_ids,
+        "downstream_port_ids": downstream_port_ids,
+        "downstream_port_id_to_router_mac_map": downstream_port_id_to_router_mac_map,
+        "upstream_port_id_to_router_mac_map": upstream_port_id_to_router_mac_map,
+        "acl_table_ports": acl_table_ports_per_dut
+    }
+
+    return t2_information
+
+
+@pytest.fixture(scope="module")
+def setup(duthosts, ptfhost, rand_selected_dut, rand_unselected_dut, tbinfo, ptfadapter, topo_scenario):
+    """Gather all required test information from DUT and tbinfo.
+
+    Args:
+        duthosts: All DUTs belong to the testbed.
+        rand_one_dut_hostname: hostname of a random chosen dut to run test.
+        tbinfo: A fixture to gather information about the testbed.
+
+    Yields:
+        A Dictionary with required test information.
+
+    """
+
+    mg_facts = rand_selected_dut.get_extended_minigraph_facts(tbinfo)
+    topo = tbinfo["topo"]["type"]
+
+    vlan_ports = []
+    vlan_mac = None
+    # Need to refresh below constants for two scenarios of M0
+    global DOWNSTREAM_DST_IP, DOWNSTREAM_IP_TO_ALLOW, DOWNSTREAM_IP_TO_BLOCK
+
+    if topo == "mx":
+        DOWNSTREAM_DST_IP = DOWNSTREAM_DST_IP_VLAN
+        DOWNSTREAM_IP_TO_ALLOW = DOWNSTREAM_IP_TO_ALLOW_VLAN
+        DOWNSTREAM_IP_TO_BLOCK = DOWNSTREAM_IP_TO_BLOCK_VLAN
+    # Announce routes for m0 is something different from t1/t0
+    if topo_scenario == "m0_vlan_scenario":
+        topo = "m0_vlan"
+        DOWNSTREAM_DST_IP = DOWNSTREAM_DST_IP_VLAN
+        DOWNSTREAM_IP_TO_ALLOW = DOWNSTREAM_IP_TO_ALLOW_VLAN
+        DOWNSTREAM_IP_TO_BLOCK = DOWNSTREAM_IP_TO_BLOCK_VLAN
+    elif topo_scenario == "m0_l3_scenario":
+        topo = "m0_l3"
+        DOWNSTREAM_DST_IP = DOWNSTREAM_DST_IP_M0_L3
+        DOWNSTREAM_IP_TO_ALLOW = DOWNSTREAM_IP_TO_ALLOW_M0_L3
+        DOWNSTREAM_IP_TO_BLOCK = DOWNSTREAM_IP_TO_BLOCK_M0_L3
+    if topo in ["t0", "mx", "m0_vlan"]:
+        vlan_ports = [mg_facts["minigraph_ptf_indices"][ifname]
+                      for ifname in list(mg_facts["minigraph_vlans"].values())[0]["members"]]
+
+        config_facts = rand_selected_dut.get_running_config_facts()
+        vlan_table = config_facts["VLAN"]
+        vlan_name = list(vlan_table.keys())[0]
+        if "mac" in vlan_table[vlan_name]:
+            vlan_mac = vlan_table[vlan_name]["mac"]
+
+    # Get the list of upstream/downstream ports
+    downstream_ports = defaultdict(list)
+    upstream_ports = defaultdict(list)
+    downstream_port_ids = []
+    upstream_port_ids = []
+    upstream_port_id_to_router_mac_map = {}
+    downstream_port_id_to_router_mac_map = {}
+
+    # For M0_VLAN/MX/T0/dual ToR scenario, we need to use the VLAN MAC to interact with downstream ports
+    # For T1/M0_L3 scenario, no VLANs are present so using the router MAC is acceptable
+    downlink_dst_mac = vlan_mac if vlan_mac is not None else rand_selected_dut.facts["router_mac"]
+
+    if topo == "t2":
```

Comment:
Same here. It seems there is some unnecessary topo check. Can you please do a cleanup?